### DISCUSSION TITLE
Objectives of the Architecture Document

### DISCUSSION THREAD

**Discussion by tlodderstedt** [openwallet-foundation/architecture-sig#68] [General]
I would like to initiate a discussion about the objectives of the architecture document.

In my opinion, there are two alternative set of objectives (see below), and it is important to agree on what we want to achieve before we move on: 

Possible objectives:
1) The document wants to distill and document typical building blocks of a wallet solution with the aim to facilitate the work of the OWF projects and inform decisions of those projects. 

2) The document specifies the reference architecture every project must follow, feature and architecture wise.  

Note: once we have a agreed on the objectives, I would also like to discuss the scope. But let's discuss objectives first.

**Comment by tkuhrt** [comment]
In past meetings, we have discussed the following uses for this document:
1. A set of capabilities that are needed for a universal wallet (one that goes beyond identity and also supports money and objects).
2. Being able to map our existing projects to this set of capabilities. This will allow us to understand:
    1. What capabilities are supported by each project.
    2. Where each project might consider expanding in the future (either through collaboration with another project in OWF or from contributions from the community).
    3. Where we might have gaps in the projects that exist within OWF that we may want to look for specific project contributions.

With that, I would say that objective 1 is what we are hoping people will use the paper for. There is no intention to require every project to support all of these capabilities; therefore, I would say that objective 2 is out.

**Comment by davidejalexander** [comment]
This discussion thread is not part of the [OWF Project mapping work](https://github.com/openwallet-foundation/architecture-sig/discussions/78) to enable cross references between each project and the Universal Wallet Reference Architecture, please see this discussion thread where are tracking this programme of work.
